subsys/bluetooth/mesh/proxy.c

```diff
@@ -953,19 +964,23 @@ static int proxy_send(struct bt_conn *conn, const void *data,
 #endif
 
 	if (!params.attr) {
-		return 0;
+		return -ENOENT;
 	}
 
 	err = bt_gatt_notify_cb(conn, &params);
-	if (!err) {
-		atomic_inc(&pending_notifications);
+	if (!err && buf) {
+		struct bt_mesh_adv *send = BT_MESH_ADV(net_buf_ref(buf));
+
+		if (!(send->count++)) {
+			bt_mesh_adv_send_start(0, 0, send->cb, send->cb_data);
+		}
```

Comment:
If you use the `destroy` callback suggested above, it doesn't really matter how many packets are sent or whether proxy is using the buffer, this mechanism can be the same in both the advertiser and the proxy. Perhaps it can be moved to a static inline function in adv.h, and use a flag instead? 

Also, there doesn't appear to be any need for putting this in the `proxy_send` function, which is only called from `proxy_segment_and_send`. As far as I can tell, if you use the destroy callback, it would be simpler to put it in `bt_mesh_proxy_relay` and `bt_mesh_pb_gatt_send`, as you won't have to pull the `net_buf` through two layers of calls